First stable release of the VenomPrecision Application

limavfabio opened the pull request:
This pull request introduces the first stable release of the VenomPrecision application, which includes a range of new features, enhancements, and bug fixes. The application is built using React and Redux, and the development environment utilizes Vite.

Key Features:

- Login and Signup Features: Users can now create accounts, log in, and securely authenticate themselves within the VenomPrecision application.

- Product Creation and Management: Users have the ability to create new products and easily manage them within the application. They can also reserve products, specifying the desired city and date.

- Fully Responsive Design: The application is designed to be fully responsive, offering seamless user experiences across various devices, including desktop and mobile.

- Deployment on Netlify: The application has been successfully deployed on Netlify and is accessible at the following URL: https://velvety-genie-9ab88d.netlify.app/. Users can now access the application and explore its functionalities.

Additional Features:

- Toast Notifications: Toast notifications have been implemented to provide users with important updates, alerts, and messages in a non-intrusive manner.

- Loading Spinners: Loading spinners have been added to indicate the progress of certain actions or processes within the application, ensuring a smoother user experience.

- Responsive Header and Sidebar: The application now features a responsive header and sidebar, adapting seamlessly to different screen sizes and improving navigation and accessibility.

- Carousel for Main Page Items: A carousel component has been incorporated into the main page, allowing users to browse through a visually appealing presentation of all available items.

You can find the Pull Request for the back-end project here: https://github.com/limavfabio/VenomPrecision-back-end/pull/25

limavfabio committed:
Setup react-router-dom for routing

limavfabio committed:
Add linters

limavfabio committed:
Rename router

limavfabio committed:
Setup Tailwindcss

limavfabio committed:
Add husky for git hooks

limavfabio committed:
Fix linters

limavfabio committed:
Initial sidebar design

limavfabio committed:
Merge pull request #12 from limavfabio/react-router-setup

React router setup

limavfabio committed:
Initial details page design

JakariaJishan committed:
Design home page

JakariaJishan committed:
Fix linters offenses

JakariaJishan committed:
Merge pull request #13 from limavfabio/home-page

Home page

JakariaJishan committed:
Style reserve page

limavfabio committed:
Create responsive sidebar

limavfabio committed:
Merge branch 'development' into sidebar-design

limavfabio committed:
Fix linters

limavfabio committed:
Remove uneccessary Husky package

JakariaJishan committed:
Merge pull request #14 from limavfabio/reserve

Style reserve page

limavfabio committed:
Merge branch 'development' into sidebar-design

limavfabio committed:
Merge pull request #15 from limavfabio/sidebar-design

Sidebar design

JakariaJishan committed:
Create form for add item

limavfabio committed:
Style Details Page

JakariaJishan committed:
Add swiper to home page

JakariaJishan committed:
Fix linter errors

limavfabio committed:
Merge pull request #16 from limavfabio/home-page-slider

Home page slider

JakariaJishan committed:
Update homepage syntext

limavfabio committed:
pnpm install

limavfabio committed:
Fix sidebar per route

limavfabio committed:
Merge branch 'sidebar-design' into details-page

limavfabio committed:
Merge branch 'development' into details-page

limavfabio committed:
Merge branch 'home-page-slider' into details-page

limavfabio committed:
Style details Page

limavfabio committed:
Add props and defaults

limavfabio committed:
Merge pull request #17 from limavfabio/details-page

Details page

limavfabio committed:
Set z index for sidebar to render above elements

limavfabio committed:
Setup redux

limavfabio committed:
Setup data fetching on main page

limavfabio committed:
Merge pull request #18 from limavfabio/setup-redux

Setup redux

limavfabio committed:
Form adds a new item

Using builtin fetch function

JakariaJishan committed:
Configure details page with database data

JakariaJishan committed:
Add react calender to reserve page

limavfabio committed:
Login page sets username on redux store

limavfabio committed:
Persist username on localStorage

limavfabio committed:
Remove unneccessary connect

limavfabio committed:
Only store username if user exists

limavfabio committed:
Handle login error and success

Render error when user don't exist and
redirect to root page when logged succesfully

limavfabio committed:
Implement logout button on sidebar

Through a custom hook useLogout

limavfabio committed:
Fix linters

limavfabio committed:
Merge pull request #19 from limavfabio/post_to_db

User Login and Logout

JakariaJishan committed:
Implement post method for reservations

JakariaJishan committed:
Fix merging

JakariaJishan committed:
Refactor the reservation

JakariaJishan committed:
Refactor slicer for reservations

limavfabio committed:
Fetch product details for each product

limavfabio committed:
setUser now also sets the id

limavfabio committed:
Fix fetch userId from store

JakariaJishan committed:
Fetch data for all list of reservations

JakariaJishan committed:
Merge pull request #20 from limavfabio/details-integration

Details integration

limavfabio committed:
Change url to production

limavfabio committed:
Render product details

limavfabio committed:
Merge branch 'auth-features' into reservations

limavfabio committed:
Add userId to redux store

By fetching userId on login and storing to redux

limavfabio committed:
linters

limavfabio committed:
Merge pull request #21 from limavfabio/reservations

Reservations

limavfabio committed:
Decide on routes

limavfabio committed:
Merge pull request #22 from limavfabio/fix-routes

Decide on routes

JakariaJishan committed:
Replace id with productId in details page

limavfabio committed:
miscellaneous changes

limavfabio committed:
Use localhost for development db

limavfabio committed:
Fetch product details based on parameter

limavfabio committed:
Set localhost db for development

limavfabio committed:
Fix redirect link

limavfabio committed:
Redirect a user to login page if not logged in

limavfabio committed:
Update products API endpoint for all products

limavfabio committed:
Create Signup

Component and redux store implementation

JakariaJishan committed:
Add reservations

limavfabio committed:
Merge pull request #23 from limavfabio/reservation

Reservation

limavfabio committed:
Style login and sign up components

limavfabio committed:
signup improvement

limavfabio committed:
Create products page

limavfabio committed:
linters

limavfabio committed:
Add links to sidebar

limavfabio committed:
Improve details page responsiveness

limavfabio committed:
Route based bg-lime

limavfabio committed:
Abstract SidebarLink into component

limavfabio committed:
Improve responsiveness of products table

limavfabio committed:
Item form adds item to current_user

limavfabio committed:
Disable delete button of not-owner

limavfabio committed:
Redirect to Products#Show on item creation

limavfabio committed:
Rename header

JakariaJishan committed:
Merge with development branch

limavfabio committed:
Handle logged user state on sidebar

JakariaJishan committed:
Fix styling in home and reserve page

JakariaJishan committed:
Merge branch 'fix-auth-edge-cases' into finishing-reservations

limavfabio committed:
A user can delete his items

limavfabio committed:
Merge branch 'development' into fix-auth-edge-cases

limavfabio committed:
fix desktop sidebar logout button

limavfabio committed:
fix linters

limavfabio committed:
Merge pull request #24 from limavfabio/fix-auth-edge-cases

Fix auth edge cases

limavfabio committed:
Change URL to be the deployed APi

JakariaJishan committed:
Design bottom button

limavfabio committed:
Fixes on sidebar

limavfabio committed:
Merge branch 'finishing-reservations' into fix-auth-reservations-merge

limavfabio committed:
Add header and sidebar to reservations page

limavfabio committed:
Improve overflow on tables with many elements

limavfabio committed:
Fix reservations table horizontal overflow

On large screens

JakariaJishan committed:
Add Toastify

limavfabio committed:
Merge branch 'finishing-reservations' into fix-auth-reservations-merge

JakariaJishan committed:
Merge pull request #25 from limavfabio/finishing-reservations

Finishing reservations

limavfabio committed:
Merge pull request #26 from limavfabio/fix-auth-reservations-merge

Fix auth reservations merge

limavfabio committed:
Fix auto-fixable errors

JakariaJishan committed:
Fix linters errors without props validations

JakariaJishan committed:
Add toastify for add and delete product

JakariaJishan committed:
Redesign the form of list products

JakariaJishan committed:
Add icons to sidebar

limavfabio committed:
Fix sidebar consistency

limavfabio committed:
Remove unnecessary code

limavfabio committed:
Fix linters and propTypes

limavfabio committed:
Fix propTypes for product

Add shape for initialState of product

limavfabio committed:
Fix typo

from 'Calender' to 'Calendar'

limavfabio committed:
Fix propTypes for ReserveCalendar

limavfabio committed:
Fix function props for ReserveCity and ReserveCalendar

limavfabio committed:
Delete uneccessary function

limavfabio committed:
Abstract CityOption to a component

limavfabio committed:
Fix linters

limavfabio committed:
Add PropTypes for CityOption

limavfabio committed:
Fix accessibility issues on CityOption

limavfabio committed:
Merge pull request #27 from limavfabio/fix-linters

Fix linters

limavfabio committed:
Add shadows to products table

limavfabio committed:
Fix Sidebar panel width among pages

limavfabio committed:
Add layout to Form page

Sidebar and header

limavfabio committed:
Fix propTypes being required

limavfabio committed:
Improve API_URL declaration consistency

By storing URL in a variable

JakariaJishan committed:
Add jest test for reservation slices

limavfabio committed:
Merge pull request #28 from limavfabio/improve-sidebar-layout

Improve sidebar layout

JakariaJishan committed:
Merge branch 'development' into jest-test

JakariaJishan committed:
Create test case for reservation

limavfabio committed:
Add README template

limavfabio committed:
Add project authors to license

JakariaJishan committed:
Add style to home page and table add spinner

JakariaJishan committed:
Merge branch 'jest-test' of https://github.com/limavfabio/VenomPrecision-front-end into jest-test

JakariaJishan committed:
Add my name to author and license

limavfabio committed:
Add project information to README

limavfabio committed:
Fix linters

JakariaJishan committed:
Change brand font

JakariaJishan committed:
Merge branch 'jest-test' of https://github.com/limavfabio/VenomPrecision-front-end into jest-test

JakariaJishan committed:
Fix linter errors #2

JakariaJishan committed:
Change padding for mobile version

limavfabio committed:
Merge pull request #29 from limavfabio/jest-test

Jest test

limavfabio committed:
Add live demo link

limavfabio committed:
Hide add new product link when not logged in

limavfabio committed:
Change reserve page text

JakariaJishan committed:
Add min valute to calender

JakariaJishan committed:
Add min valute to calender #

limavfabio committed:
Add logo to sidebar

limavfabio committed:
Merge branch 'add-logo' of github.com:limavfabio/VenomPrecision-front-end into add-logo

limavfabio committed:
Add logo to signup and login pages

limavfabio committed:
Discover more models now links to all products

limavfabio committed:
Add logo to readme and other fixes

limavfabio committed:
Merge pull request #30 from limavfabio/add-logo

Add logo

mcrd25 commented on src/components/Item-form/ItemForm.jsx:
```diff
@@ -0,0 +1,142 @@
+              </label>
+              <label htmlFor="image" className="mb-2 block">
+                <p className="font-semibold">Image</p>
+                <input
+                  type="text"
+                  id="image"
+                  value={image}
+                  placeholder="https://vespaindia.com/images/classic/vxl_yellow.webp"
+                  onChange={(e) => setImage(e.target.value)}
+                  className="my-2 w-full border border-[#BFD872] px-6 py-2 focus:border-[#85a80d] focus:bg-white focus:outline-none"
+                  required
+                />
+              </label>
+              <label htmlFor="price" className="mb-2 block">
+                <p className="font-semibold">Price</p>
+                <input
+                  type="text"
+                  id="price"
+                  value={price}
+                  placeholder="Price"
+                  onChange={(e) => setPrice(e.target.value)}
+                  className="my-2 w-full border border-[#BFD872] px-6 py-2 focus:border-[#85a80d] focus:bg-white focus:outline-none"
+                  required
+                />
+              </label>
+              <button
+                type="submit"
+                className="mt-4 bg-[#97BF0F] px-6 py-2 text-white hover:bg-[#85a80d] focus:bg-[#6d8a0a]"
+              >
+                Submit
+              </button>
+              <button
+                type="button"
+                className="ml-5 mt-4 bg-[#E4E5E9] px-6 py-2 text-black hover:bg-[#cacacc] focus:bg-[#9a9a9c]"
+                onClick={() => redirect(-1)}
+              >
+                Cancel
+              </button>
+            </div>
+          </form>
```
Some of these elements are reptitive. Don't forget DRY.

Consider using a subcomponent for form inputs like so:

```js
const FormInput = ({ id, value, placeholder, onChange, className, labelText }) => (
  <label htmlFor={id} className="mb-2 block">
    <p className="font-semibold">{labelText}</p>
    <input
      type="text"
      id={id}
      value={value}
      placeholder={placeholder}
      onChange={onChange}
      className={className}
      required
    />
  </label>
);

FormInput.propTypes = {
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
  labelText: PropTypes.string.isRequired,
};
```

mcrd25 reviewed (changes requested):
Hi Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- All functional requirements seem to have been met 👍 
- README is good 👍 
- Deployed live demo 👍 

## Required Changes ♻️
_Check the comments under the review._
- *  I am not seeing tests for your Front-End code - this is something we expect to be implemented for the final evaluation. Here are some resources you can use:
   *  https://reactjs.org/docs/testing.html
   * https://jestjs.io/docs/tutorial-react
   * https://www.freecodecamp.org/news/testing-react-hooks/
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- I noticed you started with jest, please do consider adding tests using jest
- There are some candidates to make some of these html components reusable.
- Seeing where you can make code better by following best practices: See the following articles
   - https://www.freecodecamp.org/news/best-practices-for-react/
   - https://betterprogramming.pub/21-best-practices-for-a-clean-react-project-df788a682fb

**Hope all is well, and that my comments were helpful. 😸**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

JakariaJishan commented:
@mcrd25 
hello mcrd25,
thanks for your suggestions,  could you provide some notable examples of which part of our website differs from the original design?

JakariaJishan committed:
Add test for Reservation

JakariaJishan committed:
Add test for productSlice

JakariaJishan committed:
Add more test for productSlice #2

JakariaJishan committed:
Create test for userslice

limavfabio committed:
Initial details test

JakariaJishan committed:
Configure react-testing cases

JakariaJishan committed:
Merge branch 'create-tests' of https://github.com/limavfabio/VenomPrecision-front-end into create-tests

limavfabio committed:
Move details test to __test__folder

JakariaJishan committed:
Create reservations test

JakariaJishan committed:
Merge branch 'create-tests' of https://github.com/limavfabio/VenomPrecision-front-end into create-tests

JakariaJishan committed:
Add reservations test #2

JakariaJishan committed:
Finally pass tests for reservations

limavfabio committed:
Fix tests for Details

limavfabio committed:
Merge branch 'development' into create-tests

JakariaJishan committed:
Add test for Reserve

JakariaJishan committed:
Merge

limavfabio committed:
Abstract form inputs into component

JakariaJishan committed:
Add snapshots for components

JakariaJishan committed:
Merge branch 'create-tests' of https://github.com/limavfabio/VenomPrecision-front-end into create-tests

JakariaJishan committed:
Fix passing tezt

JakariaJishan committed:
Fix index.css

limavfabio committed:
Remove unused tests

limavfabio committed:
Type coerce to match propTypes

limavfabio committed:
Fix propTypes for ReserveButton

JakariaJishan committed:
Fix some linters errors

JakariaJishan committed:
Fix merging

JakariaJishan committed:
Merge branch 'create-tests' of https://github.com/limavfabio/VenomPrecision-front-end into create-tests

limavfabio committed:
Remove unused tests

limavfabio committed:
Changed initial value from string to null

This is because price is expected to be a number,
and the default value being an empty string makes some functions
expect a string when it should really be a number.

limavfabio committed:
Add propTypes for FormInput

JakariaJishan committed:
Recover two delelted files

limavfabio committed:
Delete unused component

JakariaJishan committed:
Merge branch 'create-tests' of https://github.com/limavfabio/VenomPrecision-front-end into create-tests

JakariaJishan committed:
Merge branch 'create-tests' of https://github.com/limavfabio/VenomPrecision-front-end into create-tests

JakariaJishan committed:
Recover details and about test case

JakariaJishan committed:
Fix stylelint error

JakariaJishan committed:
Remove font file

limavfabio committed:
Move type coercion to child component

To avoid calling method on undefined value

limavfabio committed:
Merge branch 'create-tests' of github.com:limavfabio/VenomPrecision-front-end into create-tests

JakariaJishan committed:
Fix setCity issue

limavfabio committed:
Fix undefined prop mismatch

JakariaJishan committed:
Fix Calender issue

JakariaJishan committed:
Merge branch 'create-tests' of https://github.com/limavfabio/VenomPrecision-front-end into create-tests

limavfabio committed:
Fixed propTypes mismatch

limavfabio committed:
Merge branch 'create-tests' of github.com:limavfabio/VenomPrecision-front-end into create-tests

JakariaJishan committed:
Change the dependency of handle calender

JakariaJishan committed:
Merge branch 'create-tests' of https://github.com/limavfabio/VenomPrecision-front-end into create-tests

JakariaJishan committed:
Convert value price to string

JakariaJishan committed:
Fix Proptype mismatch

JakariaJishan committed:
Remove console log

limavfabio committed:
Fix propTypes for ReserveButton

on product.price

limavfabio committed:
Fix missing deps on useEffect hooks

limavfabio committed:
Fix FormInput

limavfabio committed:
Fix linters

limavfabio committed:
Fix created at propTypes

limavfabio committed:
Merge pull request #33 from limavfabio/create-tests

Create tests

Meltrust reviewed (approved):
#### Hi @limavfabio,


# **Wow, you did it 🎉**
![Brilliant](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
![applause](https://media.giphy.com/media/srg19CG0cKMuI/giphy.gif)

### To highlight

- Tests have been implemented✔️ 
- console.log statements have been removed✔️ 
- ItemForm.jsx has been refactored✔️ 
- After refactoring, application still works ✔️ 
- Good job, and good luck ✔️ 



Cheers and Happy coding!👏👏👏


------
